pkgs/applications/networking/instant-messengers/linphone/default.nix

```diff
@@ -53,13 +53,20 @@ in stdenv.mkDerivation rec {
   # some of the dependencies so let's use the latest commit.
   version = "unstable-2020-03-06";
 
-  # SEE THESE FOR HELP:
-  # - https://aur.archlinux.org/packages/linphone-desktop-git/
-  # - https://aur.archlinux.org/packages/linphone-git/
-
-  src = fetchurl {
-    url = "https://gitlab.linphone.org/BC/public/linphone-desktop/-/archive/971997e162558d37051f89c9c34bbc240135f704/linphone-desktop-971997e162558d37051f89c9c34bbc240135f704.tar.gz";
-    sha256 = "1cshddsgqpam5czsmlpd96is3pfcmw624fcr6lsifwxsdig1nim8";
+  # Using fetchgit instead of fetchurl has the benefit that we get the .git
+  # directory which linphone can use for determining the version information.
+  # That seems to be broken though, see:
+  #
+  #   https://gitlab.linphone.org/BC/public/linphone-desktop/issues/21
+  #
+  # But let's use this fetchgit anyway here so it's already in use when the
+  # version information parsing works and one can test it now.
+  src = fetchgit {
+    url = "https://gitlab.linphone.org/BC/public/linphone-desktop.git";
+    rev = "971997e162558d37051f89c9c34bbc240135f704";
+    sha256 = "1cdlf7m2d7izx6jnmdplzya6b2d3hflxwr07xb68cgk9zpfigj67";
+    leaveDotGit = true;
```

Comment:
I understand the reasoning, but this might lead to interesting hash-stability issues sometimes. Or maybe we'll get lucky.

Reply:
Ok, I'll use fetchurl then instead :+1: